extensions/vscode/src/multiStepInputs/newDeployment.ts

```diff
@@ -302,37 +471,101 @@ export async function newDeployment(
     return newCredentialForced(state) || newCredentialSelected(state);
   };
 
-  const hasMultipleEntryPoints = () => {
+  const hasMultiplePossibleEntryPointFiles = () => {
     return entryPointListItems.length > 1;
   };
 
+  const hasMultipleContentTypesForSelectedEntryPoint = () => {
+    return inspectionResults.length > 1;
+  };
+
   const getStepInfo = (
     stepId: string,
     multiStepState: MultiStepState,
   ): stepInfo | undefined => {
     const step = steps[stepId];
     if (!step) {
-      return undefined;
+      // if we have not covered the step, then don't number it.
+      return {
+        step: 0,
+        totalSteps: 0,
+      };
     }
     if (newCredentialForced(multiStepState)) {
-      if (hasMultipleEntryPoints()) {
-        return step.noCredentials.multipleEntryPoints;
+      if (hasMultiplePossibleEntryPointFiles()) {
+        if (hasMultipleContentTypesForSelectedEntryPoint()) {
+          return step.noCredentials.multipleEntryPoints.multipleContentTypes;
+        }
+        return step.noCredentials.multipleEntryPoints.singleContentType;
       }
       return step.noCredentials.singleEntryPoint;
     }
     if (newCredentialSelected(multiStepState)) {
-      if (hasMultipleEntryPoints()) {
-        return step.newCredentials.multipleEntryPoints;
+      if (hasMultiplePossibleEntryPointFiles()) {
+        if (hasMultipleContentTypesForSelectedEntryPoint()) {
+          return step.newCredentials.multipleEntryPoints.multipleContentTypes;
+        }
+        return step.newCredentials.multipleEntryPoints.singleContentType;
       }
       return step.newCredentials.singleEntryPoint;
     }
     // else it has to be existing credential selected
-    if (hasMultipleEntryPoints()) {
-      return step.existingCredentials.multipleEntryPoints;
+    if (hasMultiplePossibleEntryPointFiles()) {
+      if (hasMultipleContentTypesForSelectedEntryPoint()) {
+        return step.existingCredentials.multipleEntryPoints
+          .multipleContentTypes;
+      }
+      return step.existingCredentials.multipleEntryPoints.singleContentType;
     }
     return step.existingCredentials.singleEntryPoint;
   };
 
+  const getConfigurationInspectionQuickPicks = (
+    relEntryPoint: EntryPointPath,
+  ) => {
+    return new Promise<QuickPickItemWithIndex[]>(async (resolve, reject) => {
+      const inspectionListItems: QuickPickItemWithIndex[] = [];
+
+      try {
+        const python = await getPythonInterpreterPath();
+        const dir = path.dirname(relEntryPoint);
+        const entryPointFile = path.basename(relEntryPoint);
```

Comment:
Looks like the issue is coming up here.

In a single folder workspace like `fastapi-simple` `entrypointFile = 'simple.py'` and `dir = '.'` which works. But with `sample-content` open I get those same results which causes the inspect API request to fail since `simple.py` is not in `sample-content` at the `dir=.` level